go/teams/chain.go

```diff
@@ -476,26 +476,40 @@ func (t *TeamSigChainState) inform(u keybase1.UserVersion, role keybase1.TeamRol
 	}
 }
 
-func (t *TeamSigChainState) informNewInvite(i keybase1.TeamInvite) {
-	t.inner.ActiveInvites[i.Id] = i
+func (t *TeamSigChainState) informNewInvite(i keybase1.TeamInvite, teamSigMeta keybase1.TeamSignatureMetadata) {
+	t.inner.InviteMetadatas[i.Id] = keybase1.NewTeamInviteMetadata(i, teamSigMeta)
 }
 
-func (t *TeamSigChainState) informCanceledInvite(i keybase1.TeamInviteID) {
-	delete(t.inner.ActiveInvites, i)
-	delete(t.inner.ObsoleteInvites, i)
+func (t *TeamSigChainState) informCanceledInvite(i keybase1.TeamInviteID,
+	cancelTeamSigMeta keybase1.TeamSignatureMetadata) {
+	inviteMD, ok := t.inner.InviteMetadatas[i]
+	if !ok {
+		return
+	}
+	inviteMD.Status = keybase1.NewTeamInviteMetadataStatusWithCancelled(keybase1.TeamInviteMetadataCancel{
+		TeamSigMeta: cancelTeamSigMeta,
+	})
+	t.inner.InviteMetadatas[i] = inviteMD
 }
 
-func (t *TeamSigChainState) informCompletedInvite(i keybase1.TeamInviteID) {
-	delete(t.inner.ActiveInvites, i)
-	delete(t.inner.ObsoleteInvites, i)
+func (t *TeamSigChainState) informCompletedInvite(i keybase1.TeamInviteID,
+	completeTeamSigMeta keybase1.TeamSignatureMetadata) {
+	inviteMD, ok := t.inner.InviteMetadatas[i]
+	if !ok {
+		return
+	}
+	inviteMD.Status = keybase1.NewTeamInviteMetadataStatusWithCompleted(keybase1.TeamInviteMetadataCompleted{
+		TeamSigMeta: completeTeamSigMeta,
+	})
+	t.inner.InviteMetadatas[i] = inviteMD
 }
 
-func (t *TeamSigChainState) findAndObsoleteInviteForUser(uid keybase1.UID) {
-	for id, invite := range t.inner.ActiveInvites {
-		if inviteUv, err := invite.KeybaseUserVersion(); err == nil {
+func (t *TeamSigChainState) findAndObsoleteActiveInviteForUser(uid keybase1.UID) {
+	for _, inviteMD := range t.ActiveInvites() {
```

Comment:
This code path will be a bit unfortunate: 
- `obsoleteActiveInvites` takes `chainRoleUpdates`, loops over each UV in there and calls `findAndObsoleteActiveInviteForUser`
- each `findAndObsoleteActiveInviteForUser` calls `ActiveInvites` which loops over all invites to filter out inactive invites

it might be better to fold this entire function into `obsoleteActiveInvites`, get ActiveInvites once (should be safe because roleUpdates cannot have duplicate UIDs), and maybe even make a set of uids to speed up searching for invites to obsolete.

Reply:
`useInvites` will suffer from the same fate - filtering for Active Invites for every Used Invites pair